src/components/open-media-button.js

```diff
@@ -0,0 +1,36 @@
+const hubsRegex = /https?:\/\/(hubs.local(:\d+)?|(smoke-)?hubs.mozilla.com)\/(\w+)\/?\S*/g;
+const isHubsUrl = hubsRegex.test.bind(hubsRegex);
+
+AFRAME.registerComponent("open-media-button", {
+  init() {
+    this.label = this.el.querySelector("[text]");
+
+    this.updateSrc = () => {
+      this.src = this.targetEl.components["media-loader"].data.src;
+      this.label.setAttribute("text", "value", isHubsUrl(this.src) ? "visit" : "open link");
+    };
+
+    this.onClick = () => {
+      const directNavigate = isHubsUrl(this.src);
+      if (directNavigate) {
+        location.href = this.src;
+      } else {
+        window.open(this.src);
+      }
+    };
+
+    NAF.utils.getNetworkedEntity(this.el).then(networkedEl => {
+      this.targetEl = networkedEl;
```

Comment:
This could be above onClick but I think its a race condition for it to be above updateSrc since this can resolve synchronously in some cases I believe